site/src/github/comparison_summary.rs

```diff
@@ -57,26 +58,44 @@ pub async fn post_finished(ctxt: &SiteCtxt) {
             assert_eq!(completed, queued_commit);
 
             let is_master_commit = master_commits.contains(&queued_commit.sha);
-            post_comparison_comment(ctxt, queued_commit, is_master_commit).await;
+            if let Err(error) = post_comparison_comment(ctxt, queued_commit, is_master_commit).await
+            {
+                log::error!("Cannot post comparison comment: {error:?}");
+            }
         }
     }
 }
 
 /// Posts a comment to GitHub summarizing the comparison of the queued commit with its parent
 ///
 /// `is_master_commit` is used to differentiate messages for try runs and post-merge runs.
-async fn post_comparison_comment(ctxt: &SiteCtxt, commit: QueuedCommit, is_master_commit: bool) {
-    let client = super::client::Client::from_ctxt(
-        ctxt,
-        "https://api.github.com/repos/rust-lang/rust".to_owned(),
-    );
+async fn post_comparison_comment(
+    ctxt: &SiteCtxt,
+    commit: QueuedCommit,
+    is_master_commit: bool,
+) -> anyhow::Result<()> {
+    let client = super::client::Client::from_ctxt(ctxt, RUST_REPO_GITHUB_API_URL.to_owned());
     let pr = commit.pr;
-    let body = match summarize_run(ctxt, commit, is_master_commit).await {
+    let mut body = match summarize_run(ctxt, commit, is_master_commit).await {
         Ok(message) => message,
         Err(error) => error,
     };
 
+    body.push_str(COMMENT_MARK_SUMMARY);
     client.post_comment(pr, body).await;
+
+    let graph_client =
+        super::client::Client::from_ctxt(ctxt, RUST_REPO_GITHUB_GRAPH_URL.to_owned());
+    for comment in graph_client.get_comments(pr).await? {
+        if comment.viewer_did_author
```

Comment:
Can you explain what this `viewer_did_author` check is for? I'm not sure I fully understand it.

Reply:
My understanding is that this is true if the account/token that queries the data (so the bot) is also the original author of the comment.

Reply:
This might be worth a comment somewhere